code/modules/station_goals/bluespace_tap_events.dm

```diff
@@ -0,0 +1,157 @@
+/datum/bluespace_tap_event
+	var/name = "Unknown Anomaly (Report this to coders)"
+	var/obj/machinery/power/bluespace_tap/tap
+	/// Probability of the event not running, higher tiers being rarer
+	var/threat_level
+	var/duration
+	var/turf/tap_turf
+
+/datum/bluespace_tap_event/New(obj/machinery/power/bluespace_tap/_tap)
+	. = ..()
+	tap = _tap
+	tap_turf = get_turf(tap)
+	if(!tap)
+		stack_trace("a /datum/bluespace_tap_event was called without an involved bluespace tap.")
+		return
+	if(!istype(tap))
+		stack_trace("a /datum/bluespace_tap_event was called with (name: [tap], type: [tap.type]) instead of a bluespace tap!")
+		return
+
+/datum/bluespace_tap_event/proc/start_event()
+	on_start()
+	alert_engi()
+	tap.investigate_log("event [src] has been triggered", "bluespace_tap")
```

Comment:
Make a parent type that shares stuff with sm events please. These share a lot of common code.
```
/datum/engi_event/bluespace_tap
/datum/engi_event/supermatter
```